atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/service/AsyncTransactionService.java

```diff
@@ -47,5 +48,40 @@ public interface AsyncTransactionService {
      * @return {@link ListenableFuture} containing the map from a transaction start timestamp to transaction commit
      * timestamp, possibly missing entries if relevant transactions have not committed yet
      */
+    @Deprecated
     ListenableFuture<Map<Long, Long>> getAsync(Iterable<Long> startTimestamps);
+
+    /**
+     * Gets the transaction status associated with a given {@code startTimestamp} in a future, potentially computing
+     * it asynchronously. Known future result responses may be cached on the client-side.
+     *
+     * Future result may return unknown transaction status, which means that the transaction in question has been
+     * concluded i.e. its status cannot change.
+     *
+     * @param startTimestamp start timestamp of the transaction being looked up
+     * @return {@link ListenableFuture} containing the transaction status which can be one of the following -
+     * 1. in progress transaction had not committed yet
+     * 2. committed if the transaction has been successfully committed
+     * 3. aborted if the transaction has been successfully aborted
+     * 4. unknown if the transaction has been concluded and transactions table has been swept.
```

Comment:
This makes sense, though I think it probably belongs on the javadoc of `TransactionStatus` rather than here?

Reply:
In general I'd try to minimise how prescriptive this is in light of other parts of the system that could change. Information about other processes like TTS can be referenced perhaps in parentheses. For example, I'd probably say

```
1. IN_PROGRESS, if that the transaction with this start timestamp has started but has not committed yet,
2. COMMITTED (with a commit timestamp), if the transaction with this start timestamp has been committed,
3. ABORTED, if the transaction with this start timestamp has been aborted,
4. UNKNOWN, if the transactions table does not have information about the start timestamp. (This may, for example, be because the transaction is known to have concluded and the transactions table has been swept).
```